### PR TITLE
test: smtp credentials are valid

### PR DESCRIPTION
**Pull request by psturc**
# Description
https://issues.redhat.com/browse/INTLY-9992

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist
- [ ] This change requires a documentation update 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added a test case that will be used to verify my changes 
- [ ] Verified independently on a cluster by reviewer

### CODE REVIEW

File: test/common/sendgrid_credentials_valid.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+	username, password, host := string(smtpSecret.Data["username"]), string(smtpSecret.Data["password"]), string(smtpSecret.Data["host"])
+
+	if host != "smtp.sendgrid.net" {
+		t.Skip("Skipping the test, because the credentials were not set by a Sendgrid service")
+	}
+
+	// Test if SMTP credentials are valid using smtp.Auth method
+	auth := smtp.PlainAuth("", username, password, host)
+	client, err := smtp.Dial(host + ":587")
```
Comment (KevFan) [inline test/common/sendgrid_credentials_valid.go:26]: Should we use `smtpSecret.Data["port"]` instead :thinking:

**Review by KevFan — COMMENTED** [review]
Looks good :+1:  Verified

Just have a small comment and question

Should the test case be marked as automated ? :thinking: 
* https://github.com/integr8ly/integreatly-operator/blob/master/test-cases/tests/installation/a26-verify-sendgrid-credentials-are-configured-properly.md

File: test/common/sendgrid_credentials_valid.go (hunk 1 above)
Comment (psturc) [reply test/common/sendgrid_credentials_valid.go]: thanks, totally missed it

**Comment by psturc** [comment]
@KevFan thanks for a review, I addressed your comments in a follow-up commit

**Comment by KevFan** [comment]
\lgtm

**Comment by psturc** [comment]
hey @pmccarthy @KevFan would you mind adding a lgtm label to this PR again? I had to rebase it due to failing tests. thanks

**Pull request merged**